openedx/core/djangoapps/discussions/models.py

```diff
@@ -175,7 +176,7 @@ def __str__(self):
         )
 
     @classmethod
-    def is_enabled(cls, context_key) -> bool:
+    def is_enabled(cls, context_key: CourseKey) -> bool:
```

Comment:
This isn't perfect.

We refer to the variable as a `context_key` in some places, `course_key` in others;
only `CourseKey` is supported for now.

This change at least codifies the supported type w/ the type-hint, though it conflicts with the name, for now.
Alternatively, we could rename this to `course_key`... and then rename it back later if/when we add support.

Anyway, this seemed like the easiest change to at least enforce this in code somehow, beyond just documentation/tribal knowledge.

Reply:
Yeah I don't like the disparity in naming but I don't have an opinion on which one would be more "right" to use. This change seems easy & reasonable, but going to defer to David/Kshitij

Reply:
@xitij2000 @davidjoy Thoughts? ^^

Reply:
I think we can just change the type to LearningContextKey and raise a NotImplemented error in case it's a non-course context key. It seems like that aligns better with the current state of this code. 

That said `StackedConfigurationModel` seems to be tied to course keys for now, so that will probably also need to be changed at some point to support learning context keys.

Reply:
I don't feel strongly, as it sounds like there'll be some movement here in the future and it seems easy to update later, yah?

Reply:
> I don't feel strongly, as it sounds like there'll be some movement here in the future and it seems easy to update later, yah?

Yeah. So given that this is as-is only partially addresses the discrepancy, I'm going to punt on further changes for now. I consider this a win w/ the accurate type-hinting.